### DISCUSSION TITLE
users table may contain login names which don't exist as postgresql roles

### DISCUSSION THREAD

**Discussion by ehuelsmann** [ledgersmb/LedgerSMB#5820] [Design]
Especially after migration, the users table is likely to contain user entries which don't have an associated database role. This leads to application errors when trying to save new roles.

The least we can do is indicate the role doesn't exist on the database level and offer the admin the option to create the user being shown in the roles screen.

**Comment by ehuelsmann** [comment]
This should be part of the system validations in setup.pl.

**Reply by neilt** [reply]
Are automatically validations run during the migration?  If so, that seems the correct place for it.

**Reply by ehuelsmann** [reply]
Not automatically, but we do have consistency checks: #9067

**Reply by neilt** [reply]
Could this be made into a workflow?

**Reply by ehuelsmann** [reply]
Probably? What would the workflow look like?

**Comment by ehuelsmann** [comment]
We have a consistency check in place now. That addresses at least the issue that this problem was not visible to the use.

**Comment by ehuelsmann** [comment]
Marking closed, but please do reopen when adding the workflow concept description.
